src/boot/AppEventHandlers.js

```diff
@@ -20,16 +20,36 @@ import {
 } from '../actions';
 
 /**
- * Part of the type of upstream's `NetInfo`.
- *
- * Based on docs: https://facebook.github.io/react-native/docs/netinfo
- *
- * TODO move this to a libdef, and/or get an explicit type into upstream.
+ * Part of the interface from react-native-netinfo.
+ * https://github.com/react-native-community/react-native-netinfo/tree/v3.2.1
  */
-type ConnectionInfo = {|
-  type: 'none' | 'wifi' | 'cellular' | 'unknown' | 'bluetooth' | 'ethernet' | 'wimax',
-  effectiveType: '2g' | '3g' | '4g' | 'unknown',
-|};
+// TODO: upgrade to 4.x.x so that we can use the `flow-typed` versions.
+// Requires RN 0.60+.
+type NetInfoStateType =
+  | 'unknown'
+  | 'none'
+  | 'cellular'
+  | 'wifi'
+  | 'bluetooth'
+  | 'ethernet'
+  | 'wimax'
+  | 'vpn'
+  | 'other';
+type NetInfoConnectedDetails = {
+  isConnectionExpensive: boolean,
+};
```

Comment:
Nope; it should be `{| exact |}`, per [the source](https://github.com/react-native-community/react-native-netinfo/blob/master/src/internal/types.ts).

... although, now that I say that, I'm not sure what `{| v: T |} & {| w: U |}` should be. Taken literally, it's `Void`.

Reply:
> Nope; it should be `{| exact |}`, per [the source](https://github.com/react-native-community/react-native-netinfo/blob/master/src/internal/types.ts).
> 
> ... although, now that I say that, I'm not sure what `{| v: T |} & {| w: U |}` should be. Taken literally, it's `Void`.

Mmmm, note that is a `*.ts` file, so some things may have surprising interpretations (from a perspective of knowing Flow's syntax and semantics and not so much TS's.)

I *think* what's happening there is an `&` of two types that Flow would call *in*exact object types. So `{v: T} & {w: U}` should mean `{v: T, w: U}` (for distinct names `v`, `w`). But again in the Flow case that gets into territory where I wouldn't be entirely confident of it meaning what it seems like it should, because of that `&`.

Reply:
Obviated (for now?) by adopting `null | NetInfoConnectedDetails` as the type.